spec/services/slack_notification_service_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe SlackNotificationService do
+  describe 'initialization' do
+    subject { described_class.new({}) }
+
+    it 'raise an error if no action is set in params' do
+      expect { subject }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#send_notification' do
+    subject { described_class.new(params).send_notification }
+
+    let(:username) { Faker::Internet.username }
+    let!(:user) do
+      User.create(github_name: username, blacklisted: false)
+    end
+
+    context 'with valid params' do
+      let(:random_action) { described_class::ACTION_METHODS.keys.sample }
+      let(:channel_name) { described_class::DEFAULT_CHANNEL }
+      let(:some_language) { 'Node' }
+      let(:params) do
+        {
+          github_webhook: {
+            action: random_action,
+            label: { name: Faker::Internet.slug(words: '_') }
+          },
+          channel: channel_name,
+          pull_request: {
+            user: { login: username },
+            body: Faker::Lorem.sentence
+          },
+          repository: {
+            name: Faker::Internet.slug,
+            language: some_language
+          }
+        }
+      end
+
+      before do
+        stub_request(:post, 'https://slack.com/api/conversations.list')
+          .to_return(status: 200, body: '', headers: {})
+
+        # stub_request(:post, 'https://slack.com/api/conversations.info')
```

Comment:
why is this? 🤔

Reply:
during test construction some execution paths required the stubing of this endpoint, too. Then I simplified the scenarios and was no longer required. I kept it just in case, but maybe it's better to cleanup the code as much as possible. I can remove it.

Reply:
Yes, I think it's better

Reply:
done.